crates/biome_js_formatter/src/comments.rs

```diff
@@ -301,6 +302,75 @@ fn handle_labelled_statement_comment(
     }
 }
 
+/// Moves comments in declaration statements to behind the identifier
+///
+/// ```javascript
+/// declare module /* comment */ A {}
+/// declare /* module */ global {}
+/// ```
+fn handle_declare_comment(comment: DecoratedComment<JsLanguage>) -> CommentPlacement<JsLanguage> {
+    match (comment.enclosing_node().kind(), comment.following_node()) {
+        // Check if it is a declare statement
+        (JsSyntaxKind::TS_DECLARE_STATEMENT, Some(following)) => {
+            println!("following: {:?}", following);
+            following.children().for_each(|child| {
+                println!("\tchild: {:?}", child);
+                child.children().for_each(|grandchild| {
+                    println!("\t\tgrandchild: {:?}", grandchild);
+                });
+            });
+            match following.kind() {
+                JsSyntaxKind::TS_GLOBAL_DECLARATION => {
+                    // Global declarations have no identifier, so keep at default
+                    CommentPlacement::Default(comment)
+                }
+                JsSyntaxKind::TS_MODULE_DECLARATION
+                | JsSyntaxKind::TS_ENUM_DECLARATION
+                | JsSyntaxKind::TS_INTERFACE_DECLARATION
+                | JsSyntaxKind::TS_DECLARE_FUNCTION_DECLARATION
+                | JsSyntaxKind::TS_TYPE_ALIAS_DECLARATION => {
+                    // Move comment after the module keyword
+                    // This is the first child of the module declaration which is the identifier
+                    if following.first_child().is_some() {
+                        return CommentPlacement::leading(
+                            following.first_child().unwrap().clone(),
+                            comment,
+                        );
+                    }
+                    CommentPlacement::Default(comment)
+                }
+                JsSyntaxKind::JS_CLASS_DECLARATION => {
+                    if following.first_child().is_some()
+                        && following.first_child().unwrap().next_sibling().is_some()
+                    {
+                        return CommentPlacement::leading(
+                            following
+                                .first_child()
+                                .unwrap()
+                                .next_sibling()
+                                .unwrap()
+                                .clone(),
+                            comment,
+                        );
+                    }
+                    CommentPlacement::Default(comment)
```

Comment:
Same here, you can use `if let Some()` for `first_child`, and then `if let Some() = first_child.next_simbling()`. With this, we remove the `unwrap`.